spring-integration-rsocket/src/main/java/org/springframework/integration/rsocket/dsl/RSocketOutboundGatewaySpec.java

```diff
@@ -73,32 +87,75 @@ public RSocketOutboundGatewaySpec command(RSocketOutboundGateway.Command command
 	 * @return the spec
-	 * @see RSocketOutboundGateway#setCommandExpression(Expression)
+	 * @see RSocketOutboundGateway#setInteractionModelExpression(Expression)
+	 * @deprecated in favor of {@link #interactionModel(Function)}
 	 */
+	@Deprecated
 	public <P> RSocketOutboundGatewaySpec command(Function<Message<P>, ?> commandFunction) {
-		return command(new FunctionExpression<>(commandFunction));
+		return interactionModel(commandFunction);
+	}
+
+	/**
+	 * Configure a {@code Function} to evaluate a {@link RSocketInteractionModel}
+	 * for RSocket request type at runtime against a request message.
+	 * @param interactionModelFunction the {@code Function} to use.
+	 * @param <P> the expected request message payload type.
+	 * @return the spec
+	 * @see RSocketOutboundGateway#setInteractionModelExpression(Expression)
+	 * @since 5.2.2
+	 */
+	public <P> RSocketOutboundGatewaySpec interactionModel(Function<Message<P>, ?> interactionModelFunction) {
+		return interactionModel(new FunctionExpression<>(interactionModelFunction));
 	}
 
 	/**
 	 * Configure a SpEL expression to evaluate a {@link RSocketOutboundGateway.Command}
 	 * for RSocket request type at runtime against a request message.
 	 * @param commandExpression the SpEL expression to use.
 	 * @return the spec
-	 * @see RSocketOutboundGateway#setCommandExpression(Expression)
+	 * @see RSocketOutboundGateway#setInteractionModelExpression(Expression)
+	 * @deprecated in favor of {@link #interactionModel(String)}
 	 */
+	@Deprecated
 	public RSocketOutboundGatewaySpec command(String commandExpression) {
-		return command(PARSER.parseExpression(commandExpression));
+		return interactionModel(commandExpression);
+	}
+
+	/**
+	 * Configure a SpEL expression to evaluate a {@link RSocketInteractionModel}
+	 * for RSocket request type at runtime against a request message.
+	 * @param interactionModelExpression the SpEL expression to use.
+	 * @return the spec
+	 * @see RSocketOutboundGateway#setInteractionModelExpression(Expression)
+	 * @since 5.2.2
+	 */
+	public RSocketOutboundGatewaySpec interactionModel(String interactionModelExpression) {
+		return interactionModel(PARSER.parseExpression(interactionModelExpression));
 	}
 
 	/**
 	 * Configure a SpEL expression to evaluate a {@link RSocketOutboundGateway.Command}
 	 * for RSocket request type at runtime against a request message.
 	 * @param commandExpression the SpEL expression to use.
 	 * @return the spec
-	 * @see RSocketOutboundGateway#setCommandExpression(Expression)
+	 * @see RSocketOutboundGateway#setInteractionModelExpression(Expression)
+	 * @deprecated in favor of {@link #interactionModel(Expression)}
 	 */
+	@Deprecated
 	public RSocketOutboundGatewaySpec command(Expression commandExpression) {
-		this.target.setCommandExpression(commandExpression);
+		return interactionModel(commandExpression);
+	}
+
+	/**
+	 * Configure a SpEL expression to evaluate a {@link RSocketInteractionModel}
```

Comment:
`s/a/an/`

for **the** RSocket request...

(all occurrences)

Reply:
On it. Thank